openedx/features/course_experience/course_tools.py

```diff
@@ -0,0 +1,37 @@
+"""
+Support for course tool plugins.
+"""
+from openedx.core.lib.api.plugins import PluginManager
+
+# Stevedore extension point namespaces
+COURSE_TOOLS_NAMESPACE = 'openedx.course_tool'
+
+
+class CourseToolsPluginManager(PluginManager):
+    """
+    Manager for all of the course tools that have been made available.
+
+    All course tabs should implement `CourseTab`.
```

Comment:
As we discussed, I decided against a CourseTool base class, because that makes it hard/impossible for Django apps in other repos to provide a new course tool. I think it was a mistake that we made CourseTab so complicated that other libraries cannot provide tabs, so I wanted to avoid that.

I like the compromise that we reached together. I'll introduce a base class that is optional and which doesn't provide anything except for documentation and default methods. The documentation will include the fact that it is not necessary to use itself!